java/src/main/java/com/powsybl/python/SeriesArrayHelper.java

```diff
@@ -0,0 +1,568 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.python;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import org.apache.commons.lang3.tuple.Triple;
+import org.eclipse.collections.api.block.function.primitive.IntToIntFunction;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.IntFunction;
+import java.util.function.IntToDoubleFunction;
+import java.util.stream.Collectors;
+
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+final class SeriesArrayHelper {
+
+    static SeriesPointerArrayBuilder prepareData(Network network, PyPowsyblApiHeader.ElementType elementType) {
+        switch (elementType) {
+            case BUS:
```

Comment:
I propose to have 1 function per equipment type to continue improve readability.